components/servo/lib.rs

```diff
@@ -240,11 +247,228 @@ impl<Window> Servo<Window> where Window: WindowMethods + 'static {
         Servo {
             compositor: compositor,
             constellation_chan: constellation_chan,
+            embedder_receiver: embedder_receiver,
+        }
+    }
+
+    fn handle_window_event(&mut self, event: WindowEvent) {
+        match event {
+            WindowEvent::Idle => {
+            }
+
+            WindowEvent::Refresh => {
+                self.compositor.composite();
+            }
+
+            WindowEvent::Resize(size) => {
+                self.compositor.on_resize_window_event(size);
```

Comment:
Did you consider using a channel instead of directly using the compositor methods? Maybe the compositor and libservo won't always run in the same thread?